zetaclient/zetacore_observer.go

```diff
@@ -227,6 +234,40 @@ func (co *CoreObserver) startSendScheduler() {
 	}
 }
 
+// Bitcoin outtx is processed in a different way
+// 1. schedule one keysign on each ticker
+// 2. schedule keysign only when nonce-mark UTXO is available
```

Comment:
Is it possible the same nonce mark utxo will be available for more than one iteration? Once the txn has been broadcasted, I guess the utxo must be spent right away for this to work.

Reply:
Yeah. This is the way I had assumed that the Bitcoin node should work. The bitcoin node should be able to check double-spending on our nonce-mark UTXO. Once it get spent (confirmation == 0), it should disappear from unspent UTXO list (by query).